drivers/spi/spi_grlib_spimctrl.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (c) 2023 Frontgrade Gaisler AB
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gaisler_spimctrl
+
+#include <zephyr/drivers/spi.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_spimctrl);
+#include "spi_context.h"
+
+
+struct spimctrl_regs {
+	uint32_t conf;
+	uint32_t ctrl;
+	uint32_t stat;
+	uint32_t rx;
+	uint32_t tx;
+};
+
+#define CONF_READCMD    0x0000007f
+#define CTRL_RST        0x00000010
+#define CTRL_CSN        0x00000008
+#define CTRL_EAS        0x00000004
+#define CTRL_IEN        0x00000002
+#define CTRL_USRC       0x00000001
+#define STAT_INIT       0x00000004
+#define STAT_BUSY       0x00000002
+#define STAT_DONE       0x00000001
+
+#define SPI_DATA(dev) ((struct data *) ((dev)->data))
+
+struct cfg {
+	volatile struct spimctrl_regs *regs;
+	int interrupt;
+};
+
+struct data {
+	struct spi_context ctx;
+};
+
+static int spi_config(struct spi_context *ctx, const struct spi_config *config)
+{
+	if (config->slave != 0) {
+		LOG_ERR("More slaves than supported");
+		return -ENOTSUP;
+	}
+
+	if (SPI_WORD_SIZE_GET(config->operation) != 8) {
+		LOG_ERR("Word size must be 8");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_CS_ACTIVE_HIGH) {
+		LOG_ERR("CS active high not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_LOCK_ON) {
+		LOG_ERR("Lock On not supported");
+		return -ENOTSUP;
+	}
+
+	if ((config->operation & SPI_LINES_MASK) != SPI_LINES_SINGLE) {
+		LOG_ERR("Only supports single mode");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_TRANSFER_LSB) {
+		LOG_ERR("LSB first not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & (SPI_MODE_CPOL | SPI_MODE_CPHA)) {
+		LOG_ERR("Only supports CPOL=CPHA=0");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_OP_MODE_SLAVE) {
+		LOG_ERR("Slave mode not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_MODE_LOOP) {
+		LOG_ERR("Loopback not supported");
+		return -ENOTSUP;
+	}
+
+	ctx->config = config;
+
+	return 0;
+}
+
+static int transceive(const struct device *dev,
+				  const struct spi_config *config,
+				  const struct spi_buf_set *tx_bufs,
+				  const struct spi_buf_set *rx_bufs)
```

Comment:
weird indentation, try to configure your editor to follow linux code style.
Basically, parameters should indent below the ( + 1 space

Reply:
Thanks. The indentation has now been updated for the functions `transceive`, `transceive_async` and `release`.